lib/bundle/commands/check.rb

```diff
@@ -14,54 +16,75 @@ def reset!
         Bundle::BrewServices.reset!
       end
 
+      def exit_on_first_error?
+        Bundle::Checker.exit_on_first_error?
+      end
+
+      def output_errors?
+        ARGV.include?("--verbose")
+      end
+
       def run
-        if any_taps_to_tap? ||
-           any_casks_to_install? ||
-           any_apps_to_install? ||
-           any_formulae_to_install? ||
-           any_formulae_to_start?
+        @dsl ||= Bundle::Dsl.new(Bundle.brewfile)
+
+        checks = {
+          taps_to_tap: "Taps",
+          casks_to_install: "Casks",
+          apps_to_install: "Apps",
+          formulae_to_install: "Formulae",
+        }
+        check_method_names = checks.keys
+
+        completed_checks = []
+        errors = []
+        enumerator = exit_on_first_error? ? :any? : :map
+
+        work_to_be_done = check_method_names.send(enumerator) do |check_method|
+          check_errors = send(check_method)
+          completed_checks << check_method
+          any_errors = check_errors.any?
+          errors.concat(check_errors) if any_errors
+          any_errors
+        end
+
+        work_to_be_done = work_to_be_done.any? if work_to_be_done.class == Array
+
+        if work_to_be_done
           puts "brew bundle can't satisfy your Brewfile's dependencies."
+          unchecked_checks = (checks.keys - completed_checks)
+          completed_checks.each { |checked| puts "#{checks[checked]} were checked." }
+          unchecked_checks.each { |unchecked| puts "#{checks[unchecked]} were not checked." }
+          if output_errors?
+            errors.each { |package| puts "#{@arrow} #{package}" }
+          end
           puts "Satisfy missing dependencies with `brew bundle install`."
           exit 1
+        elsif any_formulae_to_start?
+          puts "brew bundle can't satisfy your Brewfile's dependencies."
+          puts "At least one brew formula must be started."
```

Comment:
I'd probably rather the existing wording be changed to be more ambiguous and encapsulate this.

Reply:
What would you like to see here? My goal in keeping the checks separate was to isolate `any_formulae_to_start?` from the others because its structure was significantly different and didn't really fit the mold I'd poured for the others.

I could move `"brew bundle can't satisfy your Brewfile's dependencies."` to a constant to keep that message the same across both…?

Reply:
I think if the messaging is the same that would be fine. I'd still personally prefer (but really don't feel strongly) that the services checks were done the same way as the others?

Reply:
Can we refactor the services checks to fit let in a follow-up PR?

Reply:
Sure as long as the messaging remains the same here (i.e. can't satisfy ...)